src/ExtMsg.h

```diff
@@ -109,6 +109,23 @@ struct ExtMsgReply : ExtMsg, FastAllocated<ExtMsgReply> {
 		replyHeader.documentCount++;
 	}
 
+	void setError(std::string msg, int code) {
+		// Clear if response contains any outstanding documents
+		documents.clear();
+		replyHeader.documentCount = 0;
+
+		// Set query failure flag
+		addResponseFlag(2);
+
+		// clang-format off
+		addDocument(BSON("ok" << 0 <<
+			             "$err" << msg <<
```

Comment:
Also need to set the `errmsg` field since in some cases this is expected.

Reply:
Yeah, I have considered that. It seems like `$err` is the new norm. I want to leave out `errmsg` and fix when it raises the issue. We have to understand where `errmsg` needed.